src/figuresClasses.ts

```diff
@@ -1,19 +1,102 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+// = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.Triangle;
+
+    if (this.isValidTriangle()) {
+      throw new Error('You cannot make a triangle with such sides');
+    }
+  }
 
-export class Circle {
+  private isValidTriangle(): boolean {
+    const conditions = [
+      this.a <= 0,
+      this.b <= 0,
+      this.c <= 0,
+      this.a + this.b <= this.c,
+      this.b + this.c <= this.a,
+      this.c + this.a <= this.b,
+    ];
 
+    return conditions.some((condition) => condition);
```

Comment:
```suggestion
    return !conditions.some(Boolean);
```

Reply:
If the method has a name `isValidTriangle` - we should return true if it's valid

Reply:
Good point, fixed